Fix Connection.description migration for MySQL8

potiuk opened the pull request:
Due to a bug in Breeze initialization code, we were always running
against Postgres 9.6 and MySQL 5.7, even when the matrix selected
something else.
    
(We were overwriting the POSTGRES_VERSION and MYSQL_VERSION environment
variables in initialization code)
    
There was a connected problem that hidden the utf8mb4 problem as we
migrated to MySQL client 8, where utf8mb4 in mysql client 8
was silently falling back to latin 1 on database creation time
and we have not detected that.
    
We also fix uf8mb4 migration problem that was hidden due to those
missing tests.

Fixes https://github.com/apache/airflow/issues/12588

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

ashb commented on tests/operators/test_generic_transfer.py:
```diff
@@ -55,12 +55,12 @@ def tearDown(self):
                 task_id='test_m2m',
                 preoperator=[
                     "DROP TABLE IF EXISTS test_mysql_to_mysql",
-                    "CREATE TABLE IF NOT EXISTS test_mysql_to_mysql LIKE INFORMATION_SCHEMA.TABLES",
```
Don't think you wanted this change

potiuk replied:
I actually wanted. This test was failing on MYSQL 8 because you cannot do CREATE TABLE LIKE on INFORMATION_SCHEMA.TABLES - in MySQL8 you get an error when you try to do this

```
self = <_mysql.connection closed at 55e386a803e8>
query = b'CREATE TABLE IF NOT EXISTS test_mysql_to_mysql LIKE INFORMATION_SCHEMA.TABLES'

    def query(self, query):
        # Since _mysql releases GIL while querying, we need immutable buffer.
        if isinstance(query, bytearray):
            query = bytes(query)
        if self.waiter is not None:
            self.send_query(query)
            self.waiter(self.fileno())
            self.read_query_result()
        else:
>           _mysql.connection.query(self, query)
E           _mysql_exceptions.OperationalError: (1347, "'information_schema.TABLES' is not BASE TABLE")
```

So rather than INFORMATION_SCHEMA.TABLES I picked connection as the source of the transfer (random pick really - I was working on the connection description problem, it's small enough table to be similar to information_schema.TABLES.

ashb replied:
Ah gotcha. Cool.

ashb commented on scripts/ci/libraries/_initialization.sh:
```diff
@@ -102,10 +102,10 @@ function initialization::initialize_base_variables() {
 
     # Default MySQL versions
-    export MYSQL_VERSION=${CURRENT_MYSQL_VERSIONS[0]}
+    export MYSQL_VERSION=${MYSQL_VERSION:=${CURRENT_MYSQL_VERSIONS[0]}}
```
Hey - would you mind leaving this change out so we can merge https://github.com/apache/airflow/pull/12591 as well?

potiuk replied:
Sure. I want to do it. I just want to test it now for mysql8 as well.

ashb replied:
Thanks. And I figured, just checking.

potiuk committed:
Fix Connection.description migration for MySQL8

Due to not executing MySQL8 tests Fixed in #12591 added
description for connection table was not compatible with
MySQL8 with utf8mb4 character set.

This change adds migration and fixes the previous migration
to make it compatible.
